analyzers/src/SonarAnalyzer.Common/SymbolicExecution/Roslyn/OperationProcessors/TupleBinary.cs

```diff
@@ -0,0 +1,80 @@
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+
+using SonarAnalyzer.SymbolicExecution.Constraints;
+
+namespace SonarAnalyzer.SymbolicExecution.Roslyn.OperationProcessors;
+
+internal sealed class TupleBinary : BinaryBase<ITupleBinaryOperationWrapper>
+{
+    protected override ITupleBinaryOperationWrapper Convert(IOperation operation) =>
+        ITupleBinaryOperationWrapper.FromOperation(operation);
+
+    protected override SymbolicConstraint BoolConstraintFromOperation(ProgramState state, ITupleBinaryOperationWrapper operation, bool isLoopCondition, int visitCount)
+    {
+        var isUnknown = false;
+        foreach (var pair in TupleElements(operation.LeftOperand, operation.RightOperand))
+        {
+            var areEqual = BinaryConstraint(BinaryOperatorKind.Equals, state[pair.Left], state[pair.Right], isLoopCondition, visitCount);
+            if (areEqual is null)
+            {
+                isUnknown = true;
+            }
+            else if (operation.OperatorKind is BinaryOperatorKind.Equals && areEqual == BoolConstraint.False)
+            {
+                return BoolConstraint.False;
+            }
+            else if (operation.OperatorKind is BinaryOperatorKind.NotEquals && areEqual == BoolConstraint.False)
+            {
+                return BoolConstraint.True;
+            }
+        }
+        if (isUnknown)
+        {
+            return null;
+        }
+        else
+        {
+            return operation.OperatorKind == BinaryOperatorKind.Equals
+                ? BoolConstraint.True
+                : BoolConstraint.False;
+        }
+    }
+
+    protected override ProgramState LearnBranchingConstraint(ProgramState state, ITupleBinaryOperationWrapper operation, bool isLoopCondition, int visitCount, bool falseBranch)
+    {
+        if ((operation.OperatorKind is BinaryOperatorKind.Equals) ^ falseBranch)
+        {
+            foreach (var pair in TupleElements(operation.LeftOperand, operation.RightOperand))
+            {
+                state = LearnBranchingEqualityConstraint(state, pair.Left, pair.Right, operation.OperatorKind, isLoopCondition, visitCount, falseBranch);
+            }
+        }
+        return state;
+    }
+
+    private static IEnumerable<Pair> TupleElements(IOperation leftSide, IOperation rightSide) =>
+        leftSide.AsTuple() is { } leftTuple && rightSide.AsTuple() is { } rightTuple
+            ? Enumerable.Zip(leftTuple.Elements, rightTuple.Elements, (left, right) => (left, right))
+                .SelectMany(x => TupleElements(x.left, x.right))
+            : new[] { new Pair(leftSide, rightSide) };
```

Comment:
Add some UTs with more complicated nesting scenarios. Especially ones with broken syntax where not both operands have the same format to make sure we do not crash there.